File: java/src/test/java/com/penapereira/patterns/facade/FacadeTest.java
package com.penapereira.patterns.facade;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class FacadeTest {

    @Test
    void facadeDrivesTheSubsystemInOrder() {
        assertEquals("Facade.operation(): SubsystemA.operationA, SubsystemB.operationB, SubsystemC.operationC",
                new Facade().operation());
    }

    @Test
    void subsystemClassesRemainUsableDirectly() {
        assertEquals("SubsystemA.operationA", new SubsystemA().operationA());
        assertEquals("SubsystemB.operationB", new SubsystemB().operationB());
        assertEquals("SubsystemC.operationC", new SubsystemC().operationC());
    }
}

File: javascript/src/facade/facade.test.js
import assert from "node:assert/strict";
import { describe, it } from "node:test";
import { Facade } from "./facade.js";
import { SubsystemA } from "./subsystem-a.js";
import { SubsystemB } from "./subsystem-b.js";
import { SubsystemC } from "./subsystem-c.js";

describe("Facade", () => {
  it("drives the subsystem in order", () => {
    assert.equal(new Facade().operation(), "Facade.operation(): SubsystemA.operationA, SubsystemB.operationB, SubsystemC.operationC");
  });

  it("subsystem classes remain usable directly", () => {
    assert.equal(new SubsystemA().operationA(), "SubsystemA.operationA");
    assert.equal(new SubsystemB().operationB(), "SubsystemB.operationB");
    assert.equal(new SubsystemC().operationC(), "SubsystemC.operationC");
  });
});
